app/src/main/java/ch/epfl/sdp/peakar/points/POICache.java

```diff
@@ -0,0 +1,155 @@
+package ch.epfl.sdp.peakar.points;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.google.gson.Gson;
+
+import org.osmdroid.util.BoundingBox;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Class used to cache list of POIs in the surrounding
+ * The list of POI is converted to JSON, the corresponding bounding box
+ * is added to the JSON and everything is saved in the cache folder
+ * of the android device
+ */
+public class POICache {
+    //Constants
+    private static final String CACHE_FILE_NAME = "POIPointsCache.txt";
+
+    private static POICache instance;
+    /*List of POIs contained in the cache*/
+    private static ArrayList<POIPoint> cachedPOIPoints;
+    /*Bounding box saved to cache*/
+    private static BoundingBox cachedBoundingBox;
+
+    /**
+     * Private constructor: the class is a singleton
+     */
+    private POICache(){
+        cachedPOIPoints = null;
+        cachedBoundingBox = null;
+    }
+
+    /**
+     * Get singleton
+     * @return POICache instance
+     */
+    public static POICache getInstance(){
+        if(instance == null)
+            instance = new POICache();
+        return instance;
+    }
+
+    /**
+     * Save downloaded POIs and bounding box to cache file
+     */
+    public void savePOIDataToCache(ArrayList<POIPoint> cachedPOIPoints, BoundingBox cachedBoundingBox, Context context){
+        //Create a new object with all needed information to save in JSON
+        POICacheContent poiCacheContent = new POICacheContent(cachedPOIPoints,cachedBoundingBox);
+        Gson gson = new Gson();
+        //Convert to JSON
+        String serializesCache = gson.toJson(poiCacheContent);
+        saveJson(serializesCache,context);
+    }
+
+    /**
+     * Save the serialized JSON to text file
+     * @param serializedObject string in json format
+     * @param context context
+     */
+    private static void saveJson(String serializedObject,Context context){
```

Comment:
Are we going to have more cached objects that will use Json? In that case we should move this to a class for itself.

Reply:
I think the caching of the 3D map will be done in the same class so this method can stay here